Packs/SailPointIdentityNow/Integrations/SailPointIdentityNowEventCollector/SailPointIdentityNowEventCollector.py

```diff
@@ -0,0 +1,375 @@
+                f'information of the last event in this cycle: id: {last_fetched_id}, created: {last_fetched_creation_date}.')
+            remaining_events_to_fetch -= len(events)
+            demisto.debug(f'{remaining_events_to_fetch} events are left to fetch in the next calls.')
+            last_fetched_ids = get_last_fetched_ids(events)
+            all_events.extend(events)
+        else:
+            # to avoid infinite loop, if no events are fetched, or all events are duplicates, exit the loop
+            break
+
+    next_run = {'prev_id': last_fetched_id, 'prev_date': last_fetched_creation_date, 'last_fetched_ids': last_fetched_ids}
+    demisto.debug(f'Done fetching. Sum of all events: {len(all_events)}, the next run is {next_run}.')
+    return next_run, all_events
+
+
+''' HELPER FUNCTIONS '''
+
+
+def dedup(events: List[Dict], last_fetched_ids: list, last_fetched_creation_date: str) -> List[Dict]:
+    """
+    Dedupes the events fetched based on the last fetched ids and creation date
+    This process is based on the assumption that the events are sorted by creation date
+    Args:
+        events: List of events
+        last_fetched_ids: List of the last fetched ids
+        last_fetched_creation_date: The creation date of the last fetched event
+    Returns:
+        List of deduped events
+    """
+    if not last_fetched_ids or not last_fetched_creation_date:
+        demisto.debug("No last fetched ids or creation date. Skipping deduping.")
+        return events
+
+    demisto.debug(f"Starting deduping. Number of events before deduping: {len(events)},\
+        last creation date: {last_fetched_creation_date} , last fetched ids: {last_fetched_ids}")
+
+    deduped_events = []
+    for event in events:
+        if event['created'] != last_fetched_creation_date:
+            deduped_events.extend(events[events.index(event):])
+            break
+        if event['id'] in last_fetched_ids:
+            demisto.debug(f"Dropping duplicate event: {event}")
+        else:
+            deduped_events.append(event)
+
+    demisto.debug(f"Done deduping. Number of events after deduping: {len(deduped_events)}")
+    return deduped_events
+
+
+def get_last_fetched_ids(events: List[Dict]) -> List[str]:
+    """
+    Gets the ids of the last fetched events
+    Args:
+        events: List of events, sorted by creation date
+    Returns:
+        List of the last fetched ids
+    """
+    last_creation_date = events[-1]['created']
+    list_of_ids = [event['id'] for event in reversed(events) if event['created'] == last_creation_date]
+    return list_of_ids
+
+
+def add_time_and_status_to_events(events: List[Dict] | None) -> None:
+    """
+    Adds _time and _ENTRY_STATUS fields to events
+    Args:
+        events: List of events
+    Returns:
+        None
+"""
+    if events:
+        for event in events:
+            created = event.get('created')
+            if created:
+                created = parser.parse(created)
```

Comment:
```suggestion
            created = event['created']
```

I think it must be provieded.